pkg/sns/SnsConnectionFactory.php

```diff
@@ -59,6 +60,9 @@ public function __construct($config = 'sns:')
 
                 unset($config['dsn']);
             }
+            if (\array_key_exists('topic_arns', $config) && \is_string($config['topic_arns'])) {
```

Comment:
at this stage it has to be assoc array already. The string parsing should happen if dsn is passed.

Reply:
removed, not needed when using dsn parsing